raiden/network/transport/matrix/client.py

```diff
@@ -441,8 +441,7 @@ def start_listener_thread(
             exception_handler: Optional exception handler function which can
                 be used to handle exceptions in the caller thread.
         """
-        assert not self.should_listen and self.sync_worker is None, "Already running"
-        self.should_listen = True
+        assert self.sync_worker is None, "Already running"
```

Comment:
shouldn't we have an assert for the state of the `stop_event`? Because `Runnable`s have the concept of `start` and `stop`, it could happen the even is not properly reset.

Reply:
sounds reasonable. What is your proposed solution? 

`assert self.stop_event.is_set() and self.sync_worker is None, "Already running"`

Reply:
although the client is not a runnable right?

Reply:
the client not, but the transport yes, I believe the stop event is shared, no?

Reply:
AFAIK its only used within the client